### PR TITLE
Add table to ckeditor

### PR DESCRIPTION
**Pull request by luca-rath**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

This PR adds functionality and styling for tables in the ckeditor component.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,8 @@
         "sulu-security-bundle": "file:src/Sulu/Bundle/SecurityBundle/Resources/js"
     },
     "devDependencies": {
+        "@ckeditor/ckeditor5-dev-utils": "^11.0.1",
+        "@ckeditor/ckeditor5-theme-lark": "^11.1.0",
```
Comment (danrot) [inline package.json]: These dependencies should probably go as usual dependencies into the `AdminBundle`. This way it does not work if you only install the packages in `sulu-minimal`.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.css [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    --persianRed: #cf3939;
+    --pickledBluewood: #293650;
+    --roman: #d96161;
+    --scorpion: #585858;
+    --shakespeare: #52b6ca;
+    --silver: #ccc;
+    --silverChalice: #a2a2a2;
+    --stormGray: #70768b;
+    --white: #fff;
+    --wildSand: #f5f5f5;
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.css]: That's IMO a no-go... Can't we use the colors we have defined in `colors.scss` instead?

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.css [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+    /* Overrides the default font size in the theme. */
+    --ck-font-size-base: 12px;
+    --ck-icon-size: 12px;
+
+    /* Helper variables to avoid duplication in the colors. */
+    --ck-custom-background: var(--silver);
+    --ck-custom-foreground: var(--dustyGray);
+    --ck-custom-border: var(--ck-custom-background);
+    --ck-custom-white: var(--white);
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.css]: Have we defined this variables on our own?

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.css [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    --ck-color-button-action-background: var(--shakespeare);
+    --ck-color-button-action-hover-background: var(--ck-color-button-action-background);
+    --ck-color-button-action-active-background: var(--ck-color-button-action-background);
+    --ck-color-button-action-active-shadow: none;
+    --ck-color-button-action-disabled-background: var(--ck-color-button-action-background);
+    --ck-color-button-action-text: var(--ck-custom-white);
+
+    /* Button Save and Cancel */
+    --ck-color-button-save: hsl(120, 100%, 46%);
+    --ck-color-button-cancel: hsl(15, 100%, 56%);
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.css]: Do we need these buttons at all? :thinking:

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.css [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+    /* -- Overrides the default colors used by the ckeditor5-widget package. -------------------- */
+
+    --ck-color-widget-blurred-border: var(--nobel);
+    --ck-color-widget-hover-border: var(--nobel);
+    --ck-color-widget-editable-focus-background: var(--ck-custom-white);
+
+    /* -- Overrides the default colors used by the ckeditor5-link package. ---------------------- */
+
+    --ck-color-link-default: hsl(190, 100%, 75%);
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.css]: For what reasons do we use `hsl` here? Are these the default values? Do we have to write the default values? If not I would omit them. Having a lot of code that is not required can lead to very confusing situations...

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Grid/grid.scss [hunk 6]
```diff
@@ -2,7 +2,7 @@ $itemMarginRight: 20px;
 
 .grid {
     margin-right: -$itemMarginRight;
-    overflow: hidden;
+    overflow: visible;
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Grid/grid.scss]: There was actually a reason for it to be `hidden`... Why did you change it?

**Review by danrot — CHANGES_REQUESTED** [review]

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Grid/grid.scss (hunk 6 above)
Comment (luca-rath) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Grid/grid.scss]: if the text editor is the last item in the grid, that dropdown would be cut off. why should it be hidden?
screenshot

Comment (danrot) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Grid/grid.scss]: Hm, strange, I thought this was necessary to have not too much space at the end of a form... But it seems like this problem was solved in some other way in the mean time...

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.scss [hunk 7]
```diff
@@ -1,60 +1,79 @@
...
+
+        /* shadows */
+        --ck-color-shadow-inner: none;
+        --ck-color-shadow-drop: none;
+        --ck-color-button-default-active-shadow: none;
+        --ck-color-button-on-active-shadow: none;
+
+        /* toolbar */
+        --ck-ui-component-min-height: 20px;
+        --ck-color-toolbar-background: $silver;
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/CKEditor5/ckeditor5.scss]: I would still create variables like you see at the top of the old version of this file. We have decided to do this in order to let people override these styles later. This also applies to all other color references in this file.

File: webpack.config.js [hunk 8]
```diff
@@ -6,7 +6,7 @@ const webpack = require('webpack');
 const CleanObsoleteChunksPlugin = require('webpack-clean-obsolete-chunks');
 const ManifestPlugin = require('webpack-manifest-plugin');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
-
```
Comment (danrot) [inline webpack.config.js]: Please keep this new line. This way it is easier to distinguish between imports and other code.

**Review by danrot — CHANGES_REQUESTED** [review]

**Comment by danrot** [comment]
The failing tests should also be fixed with a rebase.

File: webpack.config.js (hunk 8 above)
Comment (luca-rath) [reply webpack.config.js]: i thought, that this would be fixed automatically when merging, because there are no conflicts. but i'll do the rebase.

**Comment by luca-rath** [comment]
i thought, that this would be fixed automatically when merging, because there are no conflicts. but i'll do the rebase.

**Comment by danrot** [comment]
@luca-rath It probably would, but this way I don't know if your changes make the tests fail :wink:

**Comment by luca-rath** [comment]
@danrot ok. now it's rebased.

**Review by danrot — APPROVED** [review]
